src/DI/ContainerBuilder.php

```diff
@@ -890,4 +898,13 @@ public function getServiceName($arg)
 		return $service;
 	}
 
+
+	/**
+	 * @internal
+	 */
+	public function notifyDefinitionChanged()
```

Comment:
I was worried about a performance. But it is ok... 

updated, what do you think now?